src/Bicep.Core/TypeSystem/TypeValidator.cs

```diff
@@ -56,6 +56,18 @@ public static IList<ErrorDiagnostic> GetCompileTimeConstantViolation(SyntaxBase
                 case ResourceRefType _:
                     return sourceType.TypeKind == TypeKind.Resource;
 
+                case StringLiteralType _ when sourceType is StringLiteralType:
+                    // The name *is* the escaped string value, so we must have an exact match.
+                    return targetType.Name == sourceType.Name;
+
+                case PrimitiveType _ when sourceType is StringLiteralType:
+                    // string literals can be assigned to strings
+                    return targetType.Name == LanguageConstants.String.Name;
+
+                case StringLiteralType _ when sourceType is PrimitiveType:
+                    // string literals can be assigned from strings
```

Comment:
> string literals can be assigned from strings [](start = 22, length = 45)

Does TS do this too?

Reply:
Nope, and I'm not sure why I thought this was necessary. Let me try and remove it and see what breaks.

Reply:
So there will be an issue with string parameters being assigned to an enum property of a resource. Unless, the user declared allowed values matching the enum members, they will get a type diagnostic. (Unless we flow the type in the other direction or something...)

Reply:
Yup, that's the problem I ran into. For reasons I can't recall, I decided not to convert allowedValues into an enum of string literal types in this PR. What do you think about doing that and removing this?

Reply:
Sounds good to me.

Reply:
I've coded this up and have it working, but it's a pretty large set of changes. If it's OK with you, I think I'd like to turn this into a separate PR.